src/javascript/app/pages/trade/digit_ticker.js

```diff
@@ -0,0 +1,124 @@
+    const style_offset_correction = 2;
+
+    // Calculate left margin of the peek-box against the "width" of the container
+    const calculateLeftMargin = (width) => (width - (digit_block_size * 10)) / 2;
+
+    // Calculate peek-box left offset.
+    const calculateOffset = () => {
+        const left_offset = calculateLeftMargin($container.offsetWidth);
+        return (+current_spot * digit_block_size) + left_offset - style_offset_correction;
+    };
+
+    const fail = () => {
+        $peek.classList.remove('digit-winning', 'digit-running');
+        $peek_box.classList.remove('digit-winning', 'digit-running');
+        $peek.classList.add('digit-losing');
+        $peek_box.classList.add('digit-losing');
+    };
+
+    const init = (container_id, contract_type, barrier, tick_count, status) => {
+        total_tick_count     = tick_count;
+        current_spot         = '-';
+        contract_status      = status;
+        $container           = document.querySelector(`#${container_id}`);
+        $container.innerHTML = `
+            <div class='peek-box'>
+                <div class='mask'>0/0</div>
+                <div class='peek'></div>
+            </div>
+            <div class='digits'>
+                ${array_of_digits.map(digit => `<div class='digit digit-${digit}'>${digit}</div>`).join('')}
+            </div>
+        `;
+        paintWinningNumbers(winningNumbers(contract_type, barrier));
+        observeResize();
+    };
+
+    const observeResize = () => {
+        window.onresize = () => {
+            calculateOffset();
+        };
+    };
+
+    const paintWinningNumbers = (winning_numbers) => {
+        winning_numbers.forEach(digit => {
+            const element = document.querySelector(`.digit-${digit}`);
+            element.classList.remove('digit-losing');
+            element.classList.add('digit-winning');
+        });
+    };
+
+    const success = () => {
+        $peek_box.classList.remove('digit-losing', 'digit-running');
+        $peek.classList.remove('digit-losing', 'digit-running');
+        $peek_box.classList.add('digit-winning');
+        $peek.classList.add('digit-winning');
+    };
+
+    const queryDom = () => {
+        $peek     = $container.querySelector('.peek');
+        $peek_box = $container.querySelector('.peek-box');
+        $mask     = $peek_box.querySelector('.peek-box > .mask');
+    };
+
+    const update = (current_tick_count, { quote }) => {
+        queryDom();
+        $container.classList.remove('invisible');
+        current_spot = quote.substr(-1);
+
+        $mask.innerText = `${current_tick_count}/${total_tick_count}`;
+        $peek.innerText = current_spot;
+
+        $peek_box.classList.add('digit-running');
+        $peek.classList.add('digit-running');
+
+        $peek_box.setAttribute('style', `transform: translateX(${calculateOffset()}px)`);
+
+        if (contract_status === 'won') {
+            success();
```

Comment:
I would also suggest renaming these, perhaps to markSuccess and markFailure, just to make it clear what the function is doing somehow